### PR TITLE
Register card - backend changes for endpoint

### PR DESCRIPTION
**Pull request by sdevalapurkar-bcgov**
# Description

This PR includes the following proposed change(s):

- Register card - backend changes for endpoint
- FLA-294 I(https://justice.gov.bc.ca/jira/browse/FLA-294)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- locally

### CODE REVIEW

File: src/backend/efiling-api/jag-efiling-api.yaml [hunk 1]
```diff
@@ -582,15 +582,15 @@ components:
...
         universalId:
           type: string
           format: uuid
           description: the user universal identifier
-        cardRegistered:
-          type: boolean
-          description: if the user has a card on file or not
+        clientId:
+          type: number
+          description: bambora id is client has a registered card
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/jag-efiling-api.yaml:593]:
```suggestion
          description: bambora id if client has a registered card
```

File: src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoAccountServiceImpl.java [hunk 2]
```diff
@@ -76,15 +76,17 @@ public AccountDetails createAccount(CreateAccountRequest createAccountRequest)
     }
 
     @Override
-    public void updateClient(BigDecimal clientId) {
-
-        if (clientId == null) throw new IllegalArgumentException("client identifier is required");
+    public void updateClient(AccountDetails accountDetails) {
+        if (accountDetails == null) throw new IllegalArgumentException("account details required");
+        if (accountDetails.getClientId() == null) throw new IllegalArgumentException("client id is required");
+        if (StringUtils.isBlank(accountDetails.getInternalClientNumber())) throw new IllegalArgumentException("client identifier is required");
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-cso-starter/src/main/java/ca/bc/gov/open/jag/efilingcsostarter/CsoAccountServiceImpl.java:82]:
```suggestion
        if (StringUtils.isBlank(accountDetails.getInternalClientNumber())) throw new IllegalArgumentException("internal client number is required");
```

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java [hunk 3]
```diff
@@ -1,44 +1,25 @@
...
-    private static final UUID USER_GUID_WITH_NO_CSO = UUID.randomUUID();
-    private static final UUID USER_GUID_WITH_EJB_EXCEPTION = UUID.randomUUID();
-    private static final UUID USER_GUID_WITH_MULTI_PROFILE = UUID.randomUUID();
-    public static final String DOMAIN = "Courts";
-    public static final String APPLICATION = "CSO";
-    public static final String IDENTIFIER_TYPE = "CAP";
     private static final String INTERNAL_CLIENT_NUMBER = "123";
-
+    private static final String FAIL_INTERNAL_CLIENT_NUMBER = "1234";
+    private static final String INTERNAL_CLIENT_NUMBER1 = "NOTNULL";
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java:22]: can we rename this var to describe what it is for and how it differs from `INTERNAL_CLIENT_NUMBER`?

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java [hunk 4]
```diff
@@ -90,7 +90,22 @@ public void withNullClientIdReturnIllegalArgumentException() {
 
 
         Assertions.assertThrows(IllegalArgumentException.class, () -> {
-            sut.updateClient(null);
+            sut.updateClient(AccountDetails.builder()
+                    .internalClientNumber(INTERNAL_CLIENT_NUMBER1)
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java:94]: this method is named
```withNullClientIdReturnIllegalArgumentException```
even though it is no longer `clientId` right?

Comment (TayGov) [reply src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java:94]: Client id is required for the update

**Comment by sdevalapurkar-bcgov** [comment]
LGTM! can't approve cause I opened the PR 😸

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java (hunk 3 above)
Comment (TayGov) [reply src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/csoAccountServiceImpl/UpdateClientTest.java]: I will update it to just use the existing INTERNAL_CLIENT_NUMBER. It is just there so it is not null and the client id illegal argument is the focus of the test.

**Comment by akroon3r** [comment]
First pass looks good. :)

**Pull request merged**